sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/ExtractValue.scala

```diff
@@ -92,6 +92,8 @@ object ExtractValue {
 
 trait ExtractValue extends UnaryExpression {
   self: Product =>
+
+  override def checkInputDataTypes(): TypeCheckResult = TypeCheckResult.TypeCheckSuccess
```

Comment:
Once we are here, `ExtractValue.apply` has already filter out non-complex type for us.